File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        
        int mini=INT_MAX;
        int left=0;
        int sum=0;
        for(int right=0;right<nums.size();right++){
            sum+=nums[right];
            

            while(sum>target){
                mini=min(mini,right-left+1);
                sum-=nums[left];
                left++;

            }
            if(sum>=target)
            mini=min(mini,right-left+1);
            
        }
        if(mini==INT_MAX) return 0;
        return mini;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n=nums.length;
        int left=0;

        int ans=Integer.MAX_VALUE;
        int sum=0;
        for(int right=0;right<n;right++){
            sum+=nums[right];

            while(sum>=target){
                ans=Math.min(ans,right-left+1);
                sum-=nums[left];
                left++;
                //if(sum>=target) ans=min(ans,right-left+1);
            }
        }
        if(ans==Integer.MAX_VALUE) return 0;
        return ans;
    }
}